File: web/ui.dart
import 'globals.dart' as life;
import 'dart:html';

/// Display statistics.
void displayStats(cycle) {
  // Update stats
  getGridSize().innerHtml = "${life.rows} x ${life.columns}";
  getTotalCycles().innerHtml = "Cycles: ${life.lifeCycles}";
  getLengthOfCycle().innerHtml =
      "Length of cycle: ${life.getDurationInSeconds()} second(s)";
  getCurrentCycle().innerHtml = "Cycle: ${life.cyclesSoFar}";
}

/// Draw the cell.
/// living - true for alive, false for dead.
void drawCell(int x, y, bool living) {
  draw(x * (life.drawWidth + life.border), y * (life.drawWidth + life.border),
      life.drawWidth, life.drawHeight, living);
}

LabelElement getGridSize() {
  return querySelector('#grid-size');
}

LabelElement getTotalCycles() {
  return querySelector('#total-cycles');
}

LabelElement getLengthOfCycle() {
  return querySelector('#length-of-cycle');
}

LabelElement getCurrentCycle() {
  return querySelector('#current-cycle');
}

/// Get the canvas element
CanvasElement getCanvas() {
  return querySelector('#canvas');
}

/// Clear the canvas
void clear() {
  life.context.clearRect(0, 0, life.canvasWidth, life.canvasHeight);
}

void fill() {
  life.context.fill();
}

/// Draw a rectangle
void draw(horizontalPosition, verticalPosition, drawWidth, drawHeight, living) {
  var color;
  if (living) {
    color = life.livingColor;
  } else {
    color = life.deadColor;
  }

  life.context.fillStyle = color;
  life.context
      .fillRect(horizontalPosition, verticalPosition, drawWidth, drawHeight);
}

File: web/js/ui.js

// Display statistics.
MAIN.displayStats = function(cycle) {
    // Update stats
    var stats = MAIN.rows + " x " + MAIN.columns + "<br>"
    + "Cycles: " + parseInt(MAIN.lifeCycles) + ", Length of cycle: "
    + MAIN.sleepDuration + " second(s)<br>"
    + "Cycle: " + parseInt(cycle) + "<br>"
    + "Last cycle: " + MAIN.lastCycle + ", Living cells: " + MAIN.livingCells + "<br>";

    MAIN.stats.innerHTML = stats;
}

// Draw the cell.
// living - true for alive, false for dead.
MAIN.drawCell = function(x, y, living) {
    MAIN.draw(
      x * (MAIN.drawWidth + MAIN.border),
      y * (MAIN.drawWidth + MAIN.border),
      MAIN.drawWidth, MAIN.drawHeight, living);
}

// Get the canvas element
MAIN.getCanvas = function() {
  // console.log('function', 'getCanvas');
  return document.getElementById('canvas');
}

// Clear the canvas
MAIN.clear = function() {
  // console.log('function', 'clear');
  MAIN.context.clearRect(0, 0, MAIN.canvasWidth, MAIN.canvasHeight);
}

// Draw a rectangle
MAIN.draw = function(
    horizontalPosition,
    verticalPosition,
    drawWidth,
    drawHeight,
    living) {

    var color;
    if (living) {
      color = 'orange';
    } else {
      color = 'black';
    }

    MAIN.context.fillStyle = color;
    MAIN.context.fillRect (horizontalPosition, verticalPosition,
        drawWidth, drawHeight);
}
